packages/core/src/fhir-deduplication/resources/condition.ts

```diff
@@ -0,0 +1,114 @@
+import { CodeableConcept, Condition } from "@medplum/fhirtypes";
+import {
+  ICD_10_CODE,
+  ICD_10_OID,
+  SNOMED_CODE,
+  SNOMED_OID,
+  combineTwoResources,
+  createCompositeKey,
+  getDateFromString,
+} from "../shared";
+
+/**
+ * Approach:
+ * 1. Group same Conditions based on:
+ *      - Medical codes:
+ *          - ICD-10, if possible
+ *          // TODO: Introduce SNOMED cross-walk to match SNOMED with ICD-10
+ *          - SNOMED, if possible
+ *      - Date
+ * 2. Combine the Conditions in each group into one master condition and return the array of only unique and maximally filled out Conditions
+ */
+export function deduplicateConditions(conditions: Condition[]) {
+  const { snomedMap, icd10Map, remainingConditions } = groupSameConditions(conditions);
+  return combineConditions(snomedMap, icd10Map, remainingConditions);
+}
+
+export function groupSameConditions(conditions: Condition[]): {
+  icd10Map: Map<string, Condition>;
+  snomedMap: Map<string, Condition>;
+  remainingConditions: Condition[];
+} {
+  const snomedMap = new Map<string, Condition>();
+  const icd10Map = new Map<string, Condition>();
+  const remainingConditions: Condition[] = [];
+
+  for (const condition of conditions) {
+    const date = createDateKey(condition);
```

Comment:
i would rename the function to be just getDate
I was a little confused since I thought there were two different keys when I first read it.

Reply:
well, the date is it's own attribute for the key.. but ya, ill change it